engine/runtime/src/test/java/org/enso/compiler/ExecCompilerTest.java

```diff
@@ -231,6 +248,45 @@ public void inlineReturnSignatureWithoutArguments() throws Exception {
     assertEquals("Function-return syntax can be used with 0 arguments", 23, result.asInt());
   }
 
+  /**
+   * This test demonstrates a slightly un-intuitive, but apparently needed by our rules, behaviour of `->` with ascriptions:
+   * 1. for `foo a:Integer -> Integer` it is interpreted as foo (a:Integer) -> Integer - i.e. a function taking an Integer and returning an Integer.
+   * 2. for `foo a : Integer -> Integer`, this results in a compile error currently.
+   */
+  @Test
+  public void weirdReturnTypeSignature1() throws Exception {
+    final URI uri = new URI("memory://rts.enso");
+    final Source src = Source.newBuilder("enso", """
+    from Standard.Base import Integer
+    foo a:Integer -> Integer = a+10
+    """,uri.getAuthority())
+        .uri(uri)
+        .buildLiteral();
+
+    var module = ctx.eval(src);
+    var foo = module.invokeMember(MethodNames.Module.EVAL_EXPRESSION, "foo");
+    assertEquals(11, foo.execute(1).asInt());
+  }
+
+  @Test
+  public void weirdReturnTypeSignature2() throws Exception {
+    final URI uri = new URI("memory://rts.enso");
+    final Source src = Source.newBuilder("enso", """
+    from Standard.Base import Integer
+    foo a : Integer -> Integer = a+10
+    """,uri.getAuthority())
+        .uri(uri)
+        .buildLiteral();
+
+    try {
+      var module = ctx.eval(src);
+      var foo = module.invokeMember(MethodNames.Module.EVAL_EXPRESSION, "foo");
+      fail("Compiler error was expected, but foo evaluated successfully as: " + foo);
+    } catch (PolyglotException ex) {
+      assertEquals("Compile error: The name `foo` could not be found.", ex.getMessage());
```

Comment:
`foo` cannot be found? What does the line `foo a : Integer -> Integer = a+10` stands for then?

Reply:
It results in a compile error, I think this is a problem at the parser level - the test is just to demonstrate it.

Reply:
This is what it looks like in the REPL:
```
from Standard.Base import Integer
foo a : Integer -> Integer = a+10

main =
    foo 2
```

```
X:\NBO\test.enso:2:1: error: Unexpected expression.
    2 | foo a : Integer -> Integer = a+10
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
X:\NBO\test.enso:5:5: error: The name `foo` could not be found.
    5 |     foo 2
      |     ^~~
Aborting due to 2 errors and 0 warnings.
Execution finished with an error: Compilation aborted due to errors.
```